lib/ops/opshandler/opshandler_test.go

```diff
@@ -99,12 +101,21 @@ func (s *OpsHandlerSuite) SetUpTest(c *C) {
 		Packages:     services.Packages,
 	})
 	c.Assert(err, IsNil)
-	s.webServer = httptest.NewServer(handler)
+
+	// It is important that we launch TLS server as authentication
+	// middleware on the handler expects TLS connections.
+	s.webServer = httptest.NewTLSServer(handler)
 
 	// for regular test, let's be admins, so tests
 	// won't be affected by auth issues
 	s.client, err = opsclient.NewAuthenticatedClient(
-		s.webServer.URL, s.adminUser, "admin-password")
+		s.webServer.URL, s.adminUser, "admin-password",
+		opsclient.HTTPClient(&http.Client{
+			Transport: &http.Transport{
+				TLSClientConfig: &tls.Config{
+					InsecureSkipVerify: true,
```

Comment:
Nit: I usually try to generate an x509 cert inside the test suite, so that the tests don't run with x09 verification disabled. 

https://github.com/gravitational/satellite/blob/c383450edfb3dc4354be14d141dfd004b923e4c9/agent/server_test.go#L724-L778

Reply:
Yeah, it's actually even easier as the test TLS server provides the underlying client which is already configured with proper certs/pool.